File: csharp/1051.height-checker.cs
/*
 * @lc app=leetcode id=1051 lang=csharp
 *
 * [1051] Height Checker
 */

using System.Linq;

// @lc code=start
public class Solution {
    public int HeightChecker(int[] heights) {
        return heights
            .OrderBy(x => x)
            .Zip(heights, (f, s) => f != s)
            .Count(d => d);
    }
}
// @lc code=end

File: java/1051.height-checker.java
/*
 * @lc app=leetcode id=1051 lang=java
 *
 * [1051] Height Checker
 */

// @lc code=start
class Solution {
    public int heightChecker(int[] heights) {
        int[] count = new int[101];
        for (int h : heights) {
            ++count[h];
        }
        int ret = 0, curr = 0;
        for (int h : heights) {
            while (count[curr] == 0) {
                ++curr;
            }
            if (curr != h) {
                ++ret;
            }
            --count[curr];
        }
        return ret;
    }
}
// @lc code=end

